File: AngularUI/angular-ui/src/app/core/services/language.service.ts
import { SingleResponseModel } from './../models/response/singleResponseModel';
import { apiUrl } from './../../../api';
import { ResponseModel } from './../models/response/responseModel';
import { ServiceRepository } from './service-repository';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { LanguageModel } from '../models/language/languageModel';
import { Observable } from 'rxjs';
import { ListResponseModel } from '../models/response/listResponseModel';
import { DeleteModel } from '../models/deleteModel';

@Injectable({
  providedIn: 'root',
})
export class LanguageService
  implements ServiceRepository<LanguageModel, number>
{
  constructor(private httpClient: HttpClient) {}

  add(addModel: LanguageModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'languages/add';
    return this.httpClient.post<ResponseModel>(newPath, addModel);
  }

  delete(deleteModel: DeleteModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'languages/delete';
    return this.httpClient.post<ResponseModel>(newPath, deleteModel);
  }

  update(updateModel: LanguageModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'languages/update';
    return this.httpClient.post<ResponseModel>(newPath, updateModel);
  }

  getById(id: number): Observable<SingleResponseModel<LanguageModel>> {
    let newPath = apiUrl + 'languages/getbyid?id=' + id;
    return this.httpClient.get<SingleResponseModel<LanguageModel>>(newPath);
  }

  getAll(): Observable<ListResponseModel<LanguageModel>> {
    let newPath = apiUrl + 'languages/getall';
    return this.httpClient.get<ListResponseModel<LanguageModel>>(newPath);
  }

  getByName(name: string): Observable<SingleResponseModel<LanguageModel>> {
    let newPath = apiUrl + 'languages/getbyname?name=' + name;
    return this.httpClient.get<SingleResponseModel<LanguageModel>>(newPath);
  }

  getByCode(code: string): Observable<SingleResponseModel<LanguageModel>> {
    let newPath = apiUrl + 'languages/getbycode?code=' + code;
    return this.httpClient.get<SingleResponseModel<LanguageModel>>(newPath);
  }
}

File: FlutterUI/flutter_ui/lib/core/services/language_service.dart
import 'package:flutter_ui/core/models/delete_model.dart';
import 'package:flutter_ui/core/models/language/language_model.dart';
import 'package:flutter_ui/core/models/response/list_response_model.dart';
import 'package:flutter_ui/core/models/response/response_model.dart';
import 'package:flutter_ui/core/models/response/single_response_model.dart';
import 'package:flutter_ui/core/utilities/dependency_resolver.dart';
import 'package:flutter_ui/core/utilities/service_repository.dart';

class LanguageService extends ServiceRepository<LanguageModel, int> {
  @override
  Future<ResponseModel> add(LanguageModel addModel) async {
    var response = await httpClient.post(
      "languages/add",
      body: addModel.toJson(),
    );

    return ResponseModel.fromJson(response);
  }

  @override
  Future<ResponseModel> delete(DeleteModel deleteModel) async {
    var response = await httpClient.post(
      "languages/delete",
      body: deleteModel.toJson(),
    );

    return ResponseModel.fromJson(response);
  }

  @override
  Future<ListResponseModel<LanguageModel>> getAll() async {
    var response = await httpClient.get("languages/getall");

    return ListResponseModel<LanguageModel>.fromJson(response);
  }

  @override
  Future<SingleResponseModel<LanguageModel>> getById(int id) async {
    var response = await httpClient.get(
      "languages/getbyid",
      queryParameters: {"id": id},
    );

    return SingleResponseModel<LanguageModel>.fromJson(response);
  }

  Future<SingleResponseModel<LanguageModel>> getByName(String name) async {
    var response = await httpClient.get(
      "languages/getbyname",
      queryParameters: {"name": name},
    );

    return SingleResponseModel<LanguageModel>.fromJson(response);
  }

  Future<SingleResponseModel<LanguageModel>> getByCode(String code) async {
    var response = await httpClient.get(
      "languages/getbycode",
      queryParameters: {"code": code},
    );

    return SingleResponseModel<LanguageModel>.fromJson(response);
  }

  @override
  Future<ResponseModel> update(LanguageModel updateModel) async {
    var response = await httpClient.post(
      "languages/update",
      body: updateModel.toJson(),
    );

    return ResponseModel.fromJson(response);
  }
}
